desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,263 @@
+    alpha : ndarray
+        Unique coordinate values specifying field line labels to compute on.
+    knots_per_transit : int
+        Number of points per toroidal transit at which to sample data along field
+        line. Default is 100.
+    num_transit : int
+        Number of toroidal transits to follow field line.
+        For axisymmetric devices, one poloidal transit is sufficient. Otherwise,
+        more transits will give more accurate result, with diminishing returns.
+    num_quad : int
+        Resolution for quadrature of bounce integrals. Default is 32.
+    num_pitch : int
+        Resolution for quadrature over velocity coordinate. Default 50.
+    batch : bool
+        Whether to vectorize part of the computation. Default is true.
+    num_well : int
+        Maximum number of wells to detect for each pitch and field line.
+        Default is to detect all wells, but due to limitations in JAX this option
+        may consume more memory. Specifying a number that tightly upper bounds
+        the number of wells will increase performance.
+    name : str, optional
+        Name of the objective function.
+    jac_chunk_size : int , optional
+        Will calculate the Jacobian for this objective ``jac_chunk_size``
+        columns at a time, instead of all at once. The memory usage of the
+        Jacobian calculation is roughly ``memory usage = m0 + m1*jac_chunk_size``:
+        the smaller the chunk size, the less memory the Jacobian calculation
+        will require (with some baseline memory usage). The time to compute the
+        Jacobian is roughly ``t=t0 +t1/jac_chunk_size``, so the larger the
+        ``jac_chunk_size``, the faster the calculation takes, at the cost of
+        requiring more memory. A ``jac_chunk_size`` of 1 corresponds to the least
+        memory intensive, but slowest method of calculating the Jacobian.
+        If None, it will use the largest size i.e ``obj.dim_x``.
+
+    """
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Effective ripple ε: "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        rho=np.linspace(0.5, 1, 3),
+        alpha=np.array([0]),
+        *,
+        knots_per_transit=100,
+        num_transit=10,
+        num_quad=32,
+        num_pitch=50,
+        batch=True,
+        num_well=None,
+        name="Effective ripple",
+        jac_chunk_size=None,
+    ):
+        if target is None and bounds is None:
+            target = 0.0
+
+        rho, alpha = np.atleast_1d(rho, alpha)
+        self._dim_f = rho.size
+        self._keys_1dr = [
+            "iota",
+            "iota_r",
+            "<|grad(rho)|>",
+            "min_tz |B|",
+            "max_tz |B|",
+            "R0",  # TODO: GitHub PR #1094
```

Comment:
Does this get computed on the right grid? It's supposed to be the global R0 right? not R0 of the given surface?

Reply:
It gets computed on LinearGrid right now, which isn't accurate until #1094. We can't afford the memory of using another grid just to compute that coefficient correctly

Reply:
even with #1094, wouldn't it still depend on which surfaces are used?

Reply:
It would be a function of the max rho yes

Reply:
But isnt it supposed to be the global R0? how much does it vary as a function of rho?

Reply:
Usually you want a surface near the lcfs when optimizing for this anyway. A grid with just rho=1 after #1094  will compute R0 just as a quadrature grid does.

Reply:
ok, this is fine for now but we should mark it as a todo to make sure we're using the correct R0 for this, regardless of which surfaces are being optimized

Reply:
Would you prefer that the objective obtains `R0` directly from the first mode of `R_lmn`?